include/auxiliary.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+#include <basics.h>
+#include <config.h>
+#include <seqan/bam_io.h>
+#include <seqan/seq_io.h>
+#include <iostream>
+#include <memory>
+#include <thread>
+
+struct Auxiliary{
+    std::ofstream                                  log_file;
+    CmdOptions                                     options;
+    std::vector<std::unique_ptr<seqan::BamFileIn>> long_read_file_handles;
+    seqan::BamHeader                               long_read_header;   // The bam header object needed to fill bam context
+    seqan::BamIndex<seqan::Bai>                    long_read_bai;
+    std::vector<std::unique_ptr<seqan::BamFileIn>> short_read_file_handles;
+    seqan::BamHeader                               short_read_header;  // The bam header object needed to fill bam context
+    seqan::BamIndex<seqan::Bai>                    short_read_bai;
+    std::vector<std::unique_ptr<seqan::FaiIndex>>  faidx_file_handles;
+    std::vector<seqan::BamAlignmentRecord>         polished_reads; // stores records in case info.options.output-polished-bam is true
+
+    Auxiliary(CmdOptions & options_) : options(options_) {}
```

Comment:
move the custom constructor below the default constructors please move all the constructors to the top